src/HTTPurple/Server.purs

```diff
@@ -226,17 +227,24 @@ serveInternal inputOptions maybeNodeMiddleware settings = do
       Just nodeMiddleware -> handleExtRequestWithMiddleware $ merge routingSettings { nodeMiddleware }
       Nothing -> handleRequest routingSettings
     sslOptions = { certFile: _, keyFile: _ } <$> filledOptions.certFile <*> filledOptions.keyFile
-  server <- case sslOptions of
-    Just { certFile, keyFile } ->
-      do
-        cert' <- readTextFile UTF8 certFile
-        key' <- readTextFile UTF8 keyFile
-        let sslOpts = key := keyString key' <> cert := certString cert'
-        HTTPS.createServer sslOpts handler
-    Nothing -> HTTP.createServer handler
-  listen server options onStarted
-  let closingHandler = close server
-  registerClosingHandler filledOptions.closingHandler closingHandler
+  netServer <- case sslOptions of
+    Just { certFile, keyFile } -> do
+      cert' <- readFile certFile
+      key' <- readFile keyFile
+      server <- HTTPS.createSecureServer'
+        { key: [ key' ]
+        , cert: [ cert' ]
+        }
+      server # EE.on_ HServer.requestH handler
+      pure $ HServer.toNetServer server
+    Nothing -> do
+      server <- HTTP.createServer
+      server # EE.on_ HServer.requestH handler
```

Comment:
Here, and in the HTTPS version, is where you could return an `Effect Unit` that removes the `request` listener.